src/tests/packagers/npm.test.ts

```diff
@@ -1003,4 +1003,123 @@ describe('NPM Packager', () => {
+          name: 'samchungy-a',
+          integrity:
+            'sha512-gUv/cvd9AFYvvGep0e9m1wSAf3dfnb71eri5TjtgC6N7qvJALXFaFVOkLNBHEYGEm2ZJdosXvGqr3ISZ7Yh46Q==',
+          _id: 'samchungy-a@2.0.0',
+          extraneous: false,
+          path: '/workdir/.esbuild/.build/node_modules/samchungy-a',
+          _dependencies: {
+            'samchungy-dep-a': '1.0.0',
+          },
+          devDependencies: {},
+          peerDependencies: {},
+          dependencies: {
+            'samchungy-dep-a': {
+              version: '1.0.0',
+              resolved: 'https://registry.npmjs.org/samchungy-dep-a/-/samchungy-dep-a-1.0.0.tgz',
+              name: 'samchungy-dep-a',
+              integrity:
+                'sha512-NVac5aAU+p7bsIrUTQO438vAO8MHyNILbeckhzxhadIUqGx3L9kEZ5HTqZ+XqDIRARmOU6UmFtus6Bc7q5+mWA==',
+              _id: 'samchungy-dep-a@1.0.0',
+              extraneous: false,
+              path: '/workdir/.esbuild/.build/node_modules/samchungy-dep-a',
+              _dependencies: {},
+              devDependencies: {},
+              peerDependencies: {},
+            },
+          },
+        },
+        'samchungy-b': {
+          version: '2.0.0',
+          resolved: 'https://registry.npmjs.org/samchungy-b/-/samchungy-b-2.0.0.tgz',
+          name: 'samchungy-b',
+          integrity:
+            'sha512-i42OG9FC2Py3RfbI8bBFZi3VoN7+MxM0OUvFcWrsIgqvZMUDVI4hNKHqpE6GTt07gDDqQnxlMNehbrsQLtHRVA==',
+          _id: 'samchungy-b@2.0.0',
+          extraneous: false,
+          path: '/workdir/.esbuild/.build/node_modules/samchungy-b',
+          _dependencies: {
+            'samchungy-dep-a': '2.0.0',
+          },
+          devDependencies: {},
+          peerDependencies: {},
+          dependencies: {
+            'samchungy-dep-a': {
+              version: '2.0.0',
+              resolved: 'https://registry.npmjs.org/samchungy-dep-a/-/samchungy-dep-a-2.0.0.tgz',
+              name: 'samchungy-dep-a',
+              integrity:
+                'sha512-Yp30ASjwmyLWCGhlLTqWZa8MlBeBiaaHsmxXMwwQxK/o044vhCsPeugHqhtsZq7Xiq68/TcBux/LKId6eyPNjA==',
+              _id: 'samchungy-dep-a@2.0.0',
+              extraneous: false,
+              path: '/workdir/.esbuild/.build/node_modules/samchungy-b/node_modules/samchungy-dep-a',
+              _dependencies: {},
+              devDependencies: {},
+              peerDependencies: {},
+            },
+          },
+        },
+      },
+    };
+
+    const npm7stderr: string = ''.concat(
+      ignoredNpmError,
+      '\n',
+      JSON.stringify(
+        {
+          error: {
+            code: 'TEST_ERROR_CODE',
+            summary: 'test summary',
+            detail: 'test detail',
+          },
+        },
+        null,
+        1
+      )
+    );
+
+    spawnSpy
+      .mockResolvedValueOnce({ stderr: '', stdout: '7.0.0' })
+      .mockRejectedValueOnce(
+        new utils.SpawnError('a spawn error', JSON.stringify(v7depsList), npm7stderr)
```

Comment:
wasn't sure what the error message would be in reality, so just went for a dummy one, not that it affects any logic